### ISSUE TITLE
Upgrade the clang-format version in our CI

### ISSUE THREAD

**Issue by zasdfgbnm** [NVIDIA/Fuser#481]
The current version of `clang-format` in our CI is too old
```
$ .lintbin/clang-format --version
clang-format version 12.0.0
```
And it can not understand my code. For example, it is trying to format my code from (formated by clang-format 15)
```C++
  constexpr auto opname##_defined_checker =                                 \
      [](auto x, auto y, auto z) constexpr {                                \
        if constexpr (opcheck<decltype(x)> op opcheck<decltype(y)>) {       \
          return std::is_same_v<decltype(x op y), decltype(z)>;             \
        }                                                                   \
        return false;                                                       \
      };                                                                    \
```
into
```C++
  constexpr auto opname##_defined_checker =                                 \
      [](auto x, auto y, auto z) constexpr {                                \
    if constexpr (opcheck<decltype(x)> op opcheck<decltype(y)>) {           \
      return std::is_same_v<decltype(x op y), decltype(z)>;                 \
    }                                                                       \
    return false;                                                           \
  };                                                                        \
```
which is clearly worse looking.

Looks like clang-format 12 does not understand the curly braces correctly, and is indenting the code weirdly.

**Comment by naoyam** [comment]
Seems reasonable to me.

**Comment by zasdfgbnm** [comment]
Fixed in #503
